src/data/mod_library.cpp

```diff
@@ -27,32 +27,70 @@
 #include <utility>
 
 
+static void nativeFilesystemIterate(
+  const std::filesystem::path& p,
+  std::filesystem::directory_options options,
+  std::error_code& ec,
+  Filesystem::DirectoryIterateCallbackFuncT callback,
+  void* userData) noexcept
+{
+  namespace fs = std::filesystem;
+  auto dir_it = fs::directory_iterator(p, options, ec);
+  if (ec)
+  {
+    return;
+  }
+
+  for (const auto& dir_entry : dir_it)
+  {
+    if (!callback(dir_entry.path(), userData))
+    {
+      return;
+    }
+  }
+}
+
+const Filesystem gNativeFilesystemHandle{
+  &std::filesystem::is_directory,
+  &std::filesystem::exists,
+  &nativeFilesystemIterate};
+
 namespace rigel::data
 {
 
 namespace
 {
 
-bool isNonEmptyDirectory(const std::filesystem::directory_entry& entry)
+bool isNonEmptyDirectory(
+  const std::filesystem::path& entry,
+  const Filesystem& fsHandle)
 {
   namespace fs = std::filesystem;
-
   std::error_code err;
-  if (!entry.is_directory(err) || err)
+  if (!fsHandle.f_is_directory(entry, err) || err)
   {
     return false;
   }
 
-  if (!entry.exists(err) || err)
+  if (!fsHandle.f_exists(entry, err) || err)
```

Comment:
note to self: adapt to this code base's style guide (sorry! was in a rush for a Proof of Concept)